In [2]:
import pandas as pd
import numpy as np

In [4]:
url = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df=pd.read_csv(url)
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [6]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [8]:
df['Age'] = df['Age'].fillna(df['Age'].median())

In [9]:
df=df.drop(columns=['Cabin'])

In [12]:
df['Embarked'].fillna(df['Embarked'].mode()[0], inplace=True)

In [13]:
df.isnull().sum()

PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Embarked       0
dtype: int64

In [14]:
before=df.shape[0]
df.drop_duplicates(inplace=True)
after=df.shape[0]
print("Rows before",before,"|after:",after)

Rows before 891 |after: 891


In [16]:
df["Age"]= df['Age'].astype(int)
df['Survived']=df['Survived'].astype('category')

In [18]:
df['FamilySize']=df['SibSp']+df['Parch']

In [21]:
df.to_csv("cleaned_data.csv", index=False)

# Task 5: Data Cleaning Notes (Titanic Dataset)

## 1) Dataset Loading
- Loaded Titanic dataset using `pd.read_csv()`.
- Checked first 5 rows using `df.head()` to understand columns and sample values.

## 2) Dataset Structure Check
- Used `df.info()` to check data types and missing values.
- Found missing values mainly in `Age` and `Cabin`.

## 3) Missing Values Handling
- `Age` had 177 missing values, so I filled them using the **median** to avoid outlier impact.
- `Cabin` had 687 missing values (very high), so I dropped the column because it was not reliable.

## 4) Duplicate Removal
- Checked and removed duplicate rows using `drop_duplicates()` to avoid repeated records.

## 5) Datatype Conversion
- Converted columns into correct datatypes (example: Age into integer) for accurate analysis.

## 6) Feature Engineering (New Column)
- Created a new column `FamilySize = SibSp + Parch` to represent total family members traveling together.

## 7) Final Verification
- Verified missing values again using `df.isnull().sum()` to confirm cleaning worked.

## 8) Export Cleaned Data
- Saved the cleaned dataset as `cleaned_data.csv` using `df.to_csv()` for further analysis or modeling.
